entityloader/entityloader.js

```diff
@@ -1,484 +1,202 @@
-			upBound = 1 << attemptC;
-			trying += mathUtils.random(lowBound, upBound);
-		}
 
-		if (attemptC >= config.nickRetries) return callback(names(6));
-		function done(r) {
-			result &= r;
-			if (++ct >= 3) {
-				if (result) {
-					callback(trying);
-				} else {
-					checkUser(suggestedNick, attemptC + 1, callback);
-				}
-			}
-		}
-		function checkRoomUser(type, name) {
-			core.emit(type, {
-				ref: name,
-				session: "internal-loader"
-			}, function(err, data) {
-				if (data && data.results && data.results.length > 0) {
-					done(false);
-				}  else {
-					done(true);
+	require("./userHandler.js")(core, config);
+	require("./initHandler.js")(core, config);
+	require("./queryHandler.js")(core, config);
+	core.on("room", function(action, next) {
+		loadRelatedUser(action.to, "me", action.session, function(err, user) {
+			log.d(action.id + " loading relatedUser", err, user);
+			action.user = user;
+
+			loadEntity(action.to, "room", "internal-loader", function(error, room) {
+				if (error) {
+					if (error.message === "NO_ROOM_WITH_GIVEN_ID") {
+						action.old = {};
+						action.occupants = [];
+						action.members = [];
+						next();
+					} else {
+						next(err);
+					}
+					return;
 				}
+				action.old = room;
+				next();
 			});
-		}
-		checkRoomUser("getRooms", trying);
-		checkRoomUser("getUsers", trying);
-		checkRoomUser("getUsers", "guest-" + trying);
-
-	}
-	checkUser(suggestedNick, 0, callback);
-}
-
-function generatePick(id) {
-	return 'https://gravatar.com/avatar/' + crypto.createHash('md5').update(id).digest('hex') + '/?d=retro';
-}
-
-
-
-function  loadProps(action, callback) {
-	var wait = true, userID = action.user.id;
-	log.d("Loading user content", action.user.id);
-	core.emit("getRooms", {
-		id: uid(),
-		hasOccupant: userID,
-		session: action.session
-	}, function(err, rooms) {
-		if (err || !rooms || !rooms.results || !rooms.results.length) {
-			action.occupantOf = [];
-		} else {
-			action.occupantOf = rooms.results;
-		}
-		log.d("Loading user content: occupants", rooms.results);
-		if (wait) wait = false;
-		else callback();
-	});
-	if(!utils.isGuest(userID)) {
-		core.emit("getRooms", {
-			id: uid(),
-			hasMember: userID,
-			session: action.session
-		}, function(err, rooms) {
-			if (err || !rooms || !rooms.results || !rooms.results.length) {
-				action.memberOf = [];
-			} else {
-				action.memberOf = rooms.results;
-			}
-			log("Loading user content: members", rooms.results);
-			if (wait) wait = false;
-			else callback();
 		});
-	}else{
-		action.memberOf = [];
-		if (wait) wait = false;
-		else callback();
-	}
-}
+	}, "loader");
+	
+	core.on('upload/getPolicy', function(action, next) {
+		core.emit("getUsers", {ref:"me", session: action.session}, function(err, users) {
+			if(err) return next(err);
+			if(!users || !users.results || !users.results.length) return next(new Error("USER_NOT_INITED"));
+			action.user = users.results[0];
+			next();
+		});
+	}, "loader");
+	
```

Comment:
Perhaps a loadMe function that is called from basicLoader, note and upload/getPolicy?